sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -95,7 +100,9 @@ def pull_latest_from_table_or_query(
         if created_timestamp_column:
             timestamps.append(created_timestamp_column)
         timestamp_desc_string = " DESC, ".join(timestamps) + " DESC"
-        field_string = ", ".join(join_key_columns + feature_name_columns + timestamps)
+        field_string = ", ".join(
+            set(join_key_columns + feature_name_columns + timestamps)
```

Comment:
is there a reason you made this a set? should probably call this out in the description of the PR too since it can cause a behavior change

Reply:
I was trying to reuse `pull_latest_from_table_or_query` because it's already implemented in all offline stores including extensions. But those implementations also include deduplication, which shouldn't happen for retrieving a saved dataset.
Now I see that it probably wasn't a good idea. I will use a separate method.